### DISCUSSION TITLE
How can I force refresh the page from modal

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3395] [Help]
I have page1's button1 click and open page2 as modal, everytime the modal updated data then I need to manually refresh the page1(I am using external MySQL with custom SQL query). Is there any way to force refresh the parent page from a modal page?

**Comment by mjashanks** [comment]
Unfortunately, this won't work at the moment with a custom query. It works when using a table directly.

It's a good point though, we should support it if possible.

**Reply by bmcbride** [reply]
FWIW- modal edits seem to refresh Budibase DB tables but not external SQL tables (Postgres in my case). I haven't found a way to force a table refresh from a modal edit for SQL tables. Maybe a modal close listener could help with this?

**Reply by samg666** [reply]
There's no official way to refresh the external SQL query from a Modal. But I find a workaround, the concept is use `app state` value change to trigger the query.

1. Add a parameter something like `Counter1` to your custom SQL
2. On your data provider, pass `{{ State.Counter1 }}` as a parameter to your custom SQL query
3. On the parent page, when click a button open a modal, then update state `Counter1` to `1`
4. On the modal page, when click SAVE button then change the `Counter1 ` to `2`, then it will trigger the parent page query to be run as the app state value/SQL param has changed.

This way works for me. You can try it.

![image](https://user-images.githubusercontent.com/29649651/143357907-db609e46-e22d-4f21-ad13-40eec4fbbab7.png)
